### ISSUE TITLE
[space-infix-ops] prevent adding a space to union types

### ISSUE THREAD

**Issue by simPod** [typescript-eslint/typescript-eslint#3590]
- [x] I have tried restarting my IDE and the issue persists.
- [x] I have updated to the latest version of the packages.
- [x] I have [read the FAQ](https://github.com/typescript-eslint/typescript-eslint/blob/master/docs/getting-started/linting/FAQ.md) and my problem is not listed.

How can I disable this rule from adding a space to my union types? There seems to be no config for it. Introduced here https://github.com/typescript-eslint/typescript-eslint/pull/3360

**Repro**

```JSON
{
  "rules": {
    "@typescript-eslint/space-infix-ops": "error"
  }
}
```

```ts
(a: Date|null = null) {}
```

**Expected Result**

Code should be fine

**Actual Result**

`Operator '|' must be spaced  @typescript-eslint/space-infix-ops`

**Additional Info**

How can I disable this rule from modifying my types? I want only space around `|` when it's not type definition.

**Versions**

| package                            | version |
| ---------------------------------- | ------- |
| `@typescript-eslint/eslint-plugin` | `4.28.0` |
| `@typescript-eslint/parser`        | `4.28.0` |
| `TypeScript`                       | `4.3.4` |
| `ESLint`                           | `7.29.0` |
| `node`                             | `16` |

**Comment by bradzacher** [comment]
Don't use our version then and just use the original ESLint version.

**Comment by simPod** [comment]
Why not make it configurable?

**Comment by bradzacher** [comment]
It's as configurable as the base rule

https://eslint.org/docs/rules/space-infix-ops

Which is to say not at all.
Everyone has a preference and it's not worth the time and effort adding preferences to minutely configure every single stylistic lint rule.

https://eslint.org/blog/2020/05/changes-to-rules-policies
